integration/rpc_test.go

```diff
@@ -262,23 +298,23 @@ func TestGetStateHTTP(t *testing.T) {
 	defer teardown()
 	time.Sleep(3 * time.Second) // give skaffold time to process all events
 
-	// retrieve the state via HTTP as well, and verify the result is the same
-	httpResponse, err := http.Get(fmt.Sprintf("http://localhost:%s/v1/state", httpAddr))
-	if err != nil {
-		t.Fatalf("error connecting to gRPC REST API: %s", err.Error())
-	}
-	b, err := ioutil.ReadAll(httpResponse.Body)
-	if err != nil {
-		t.Errorf("error reading body from http response: %s", err.Error())
-	}
+	success := false
 	var httpState proto.State
-	if err := json.Unmarshal(b, &httpState); err != nil {
-		t.Errorf("error converting http response to proto: %s", err.Error())
+	for i := 0; i < stateRetries; i++ {
+		httpState = retrieveHTTPState(t, httpAddr)
+		if checkState(httpState) {
+			success = true
+			break
+		}
+		time.Sleep(waitTime)
 	}
-	for _, v := range httpState.BuildState.Artifacts {
-		testutil.CheckDeepEqual(t, event.Complete, v)
+	if !success {
+		// max attempts exceeded, log errors
+		for _, v := range httpState.BuildState.Artifacts {
```

Comment:
this is very similar to `checkState` but with assertions - can we do a single assertion rather? 
And if the assertion fails - the error message could be something like "build and deploy is not complete, state: %v" ,just the `state` variable itself.

Reply:
in fact we don't even need to do the assertion - `success` contains the assertion. 
If `!success` we can just print the error message with the `BuildState` and `DeployState` - the error message is going to be more understandable as well.